src/main/scala/io/iohk/ethereum/consensus/ethash/EthashConsensus.scala

```diff
@@ -67,8 +73,6 @@ class EthashConsensus private (
           case MockedPow => MockedMiner(node)
         }
         atomicMiner.set(Some(miner))
```

Comment:
... and then set a miner. Because it is possible to set a miner in between of those.

Reply:
@dmitry-worker could you elaborate more?

Reply:
There is already PR opened that intend to fix that: https://github.com/input-output-hk/mantis/pull/777
There is no conclusion of the solution proposed there is the right one.

The change proposed here is an important fix. IMHO it is better to merge it and deal with potential concurrency issue in PR ☝️ .